port/unix/omrsysinfo.c

```diff
@@ -531,6 +576,810 @@ omrsysinfo_get_CPU_architecture(struct OMRPortLibrary *portLibrary)
+		rc = OMR_PROCESSOR_PPC_P9;
+	}
+
+	return rc;
+}
+#endif /* defined(LINUXPPC) */
+
+#if defined(AIXPPC)
+/**
+ * @internal
+ * Populates J9ProcessorDesc *desc on AIX
+ *
+ * @param[in] desc pointer to the struct that will contain the CPU type and features.
+ *
+ * @return 0 on success, -1 on failure
+ */
+static intptr_t
+getAIXPPCDescription(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc)
+{
+	/* AIX processor */
+	if (__power_rs1() || __power_rsc()) {
+		desc->processor = OMR_PROCESSOR_PPC_RIOS1;
+	} else if (__power_rs2()) {
+		desc->processor = OMR_PROCESSOR_PPC_RIOS2;
+	} else if (__power_601()) {
+		desc->processor = OMR_PROCESSOR_PPC_PWR601;
+	} else if (__power_603()) {
+		desc->processor = OMR_PROCESSOR_PPC_PWR603;
+	} else if (__power_604()) {
+		desc->processor = OMR_PROCESSOR_PPC_PWR604;
+	} else if (__power_620()) {
+		desc->processor = OMR_PROCESSOR_PPC_PWR620;
+	} else if (__power_630()) {
+		desc->processor = OMR_PROCESSOR_PPC_PWR630;
+	} else if (__power_A35()) {
+		desc->processor = OMR_PROCESSOR_PPC_NSTAR;
+	} else if (__power_RS64II()) {
+		desc->processor = OMR_PROCESSOR_PPC_NSTAR;
+	} else if (__power_RS64III()) {
+		desc->processor = OMR_PROCESSOR_PPC_PULSAR;
+	} else if (__power_4()) {
+		desc->processor = OMR_PROCESSOR_PPC_GP;
+	} else if (__power_5()) {
+		desc->processor = OMR_PROCESSOR_PPC_GR;
+	} else if (__power_6()) {
+		desc->processor = OMR_PROCESSOR_PPC_P6;
+	} else if (__power_7()) {
+		desc->processor = OMR_PROCESSOR_PPC_P7;
+	} else if (__power_8()) {
+		desc->processor = OMR_PROCESSOR_PPC_P8;
+	} else if (__power_9()) {
+		desc->processor = OMR_PROCESSOR_PPC_P9;
+	} else {
+		desc->processor = OMR_PROCESSOR_PPC_UNKNOWN;
+	}
+#if !defined(J9OS_I5_V6R1)
+	/* AIX physical processor */
+	if (__phy_proc_imp_4()) {
+		desc->physicalProcessor = OMR_PROCESSOR_PPC_GP;
+	} else if (__phy_proc_imp_5()) {
+		desc->physicalProcessor = OMR_PROCESSOR_PPC_GR;
+	} else if (__phy_proc_imp_6()) {
+		desc->physicalProcessor = OMR_PROCESSOR_PPC_P6;
+	} else if (__phy_proc_imp_7()) {
+		desc->physicalProcessor = OMR_PROCESSOR_PPC_P7;
+	} else if (__phy_proc_imp_8()) {
+		desc->physicalProcessor = OMR_PROCESSOR_PPC_P8;
+	} else {
+		desc->physicalProcessor = desc->processor;
+	}
+#else
+		desc->physicalProcessor = desc->processor;
+#endif /* !defined(J9OS_I5_V6R1) */
+	/* AIX Features */
+	if (__power_64()) {
+		setFeature(desc, OMRPORT_PPC_FEATURE_64);
+	}
+	if (__power_vmx()) {
+		setFeature(desc, OMRPORT_PPC_FEATURE_HAS_ALTIVEC);
+	}
+	if (__power_dfp()) {
+		setFeature(desc, OMRPORT_PPC_FEATURE_HAS_DFP);
+	}
+	if (__power_vsx()) {
+		setFeature(desc, OMRPORT_PPC_FEATURE_HAS_VSX);
+	}
+#if !defined(J9OS_I5_V6R1)
+	if (__phy_proc_imp_6()) {
+		setFeature(desc, OMRPORT_PPC_FEATURE_ARCH_2_05);
+	}
+	if (__phy_proc_imp_4()) {
+		setFeature(desc, OMRPORT_PPC_FEATURE_POWER4);
+	}
+#endif /* !defined(J9OS_I5_V6R1) */
+	return 0;
+}
+
+#endif /* defined(AIXPPC) */
+
+#if (defined(S390) || defined(J9ZOS390))
```

Comment:
I'm not sure but, is this if missing `|| defined(J9ZTPF))` ?

Reply:
There was no `|| defined(J9ZTPF))` in the J9 version, so I left it as is